Editor/EditorCore/PolyShapeEditor.cs

```diff
@@ -426,11 +432,18 @@ void DoPointPlacement()
                     float distanceToVertex = Mathf.Min(Vector2.Distance(mouse, ga), Vector2.Distance(mouse, gb));
 
                     if (distanceToVertex > PreferenceKeys.k_MaxPointDistanceFromControl && distanceToLine < PreferenceKeys.k_MaxPointDistanceFromControl)
+
+                    if(distanceToVertex > PreferenceKeys.k_MaxPointDistanceFromControl &&
+                       distanceToLine < PreferenceKeys.k_MaxPointDistanceFromControl)
                     {
-                        Handles.color = Color.green;
-                        Handles.DotHandleCap(-1, wp, Quaternion.identity, HandleUtility.GetHandleSize(wp) * k_HandleSize, evt.type);
+                        if(evt.type == EventType.Repaint)
+                        {
+                            Handles.color = Color.green;
```

Comment:
Normally, I would say to use a Handles.DrawingScope for color changes but the value seems to be reset to white at the end so it should be fine